src/libfuncs/mem.rs

```diff
@@ -35,39 +35,36 @@ pub fn build<'ctx, 'this>(
     selector: &MemConcreteLibfunc,
 ) -> Result<()> {
     match selector {
-        MemConcreteLibfunc::StoreTemp(info) => {
-            build_store_temp(context, registry, entry, location, helper, info)
+        MemConcreteLibfunc::AllocLocal(info) => {
+            build_alloc_local(context, registry, entry, location, helper, metadata, info)
         }
         MemConcreteLibfunc::StoreLocal(info) => {
             build_store_local(context, registry, entry, location, helper, metadata, info)
         }
-        MemConcreteLibfunc::FinalizeLocals(info) => {
-            build_finalize_locals(context, registry, entry, location, helper, info)
-        }
-        MemConcreteLibfunc::AllocLocal(info) => {
-            build_alloc_local(context, registry, entry, location, helper, metadata, info)
-        }
-        MemConcreteLibfunc::Rename(info) => {
-            build_rename(context, registry, entry, location, helper, info)
+        MemConcreteLibfunc::FinalizeLocals(info) => super::build_noop::<0, true>(
+            context,
+            registry,
+            entry,
+            location,
+            helper,
+            metadata,
+            &info.signature.param_signatures,
+        ),
+        MemConcreteLibfunc::Rename(SignatureOnlyConcreteLibfunc { signature })
+        | MemConcreteLibfunc::StoreTemp(SignatureAndTypeConcreteLibfunc { signature, .. }) => {
+            super::build_noop::<1, false>(
+                context,
+                registry,
+                entry,
+                location,
+                helper,
+                metadata,
+                &signature.param_signatures,
+            )
```

Comment:
Out of curiosity, why some noops must not update builtin counters?

Reply:
Libfuncs that receive builtins but do not use them as a builtin, for example `store_temp<Bitwise>`.